packages/common/src/make-state-datatype.js

```diff
@@ -40,9 +40,12 @@ export default function makeStateDataType(Constructor, name) {
           };
         }
 
+        // We only want to construct the new instance if we have some set of
+        // attributes (even an empty object) to base it on. This is to deal with
+        // the unexpected side effect that AmpState#unset will create a new
+        // instance.
         return {
-          val: new Constructor(newVal, {parent: this}),
-          // val: new Constructor(newVal),
+          val: newVal ? new Constructor(newVal, {parent: this}) : undefined,
```

Comment:
right; by unsetting supertoken, a new token gets created automatically, but since it doesn't ave an access token value, Token#initialize throws an exception and breaks everything.